djangoguestbook/guestbook/test_model/test.py

```diff
@@ -0,0 +1,235 @@
+		dict = {
+			'guestbook_name': self.guestbook_name,
+			'author': "test_author",
+			'content': "testing"
+		}
+		greeting = Greeting.put_from_dict(dict)
+		Greeting.delete_greeting_by_id(self.guestbook_name, greeting.key.id())
+		got_greeting = Greeting.get_greeting_by_id(self.guestbook_name, greeting.key.id())
+		assert got_greeting is None
+
+	def test_delete_greeting_wiht_right_id(self):
+		dict = {
+			'guestbook_name': self.guestbook_name,
+			'author': "test_author",
+			'content': "testing"
+		}
+		greeting = Greeting.put_from_dict(dict)
+		Greeting.delete_greeting_by_id(self.guestbook_name, greeting.key.id()+1)
+		got_greeting = Greeting.get_greeting_by_id(self.guestbook_name, greeting.key.id())
+		assert got_greeting is not None and got_greeting == greeting
+
+	def test_to_dict_with_right_data(self):
+		greeting = Greeting()
+		greeting.author = "test_author"
+		greeting.content = "testing"
+		greeting.date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.updated_by = "updated_author"
+		greeting.updated_date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.put()
+		test_dict = {
+			'id': greeting.key.id(),
+			'author': 'test_author',
+			'content': "testing",
+			'date': "2015-02-11 13:00 +0000",
+			'updated_by': "updated_author",
+			'updated_date': "2015-02-11 07:11 +0000"
+		}
+		assert test_dict == greeting.to_dict()
+
+	def test_to_dict_with_wrong_id(self):
+		greeting = Greeting()
+		greeting.author = "test_author"
+		greeting.content = "testing"
+		greeting.date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.updated_by = "updated_author"
+		greeting.updated_date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.put()
+		test_dict = {
+			'id': greeting.key.id() + 1,
+			'author': 'test_author',
+			'content': "testing",
+			'date': "2015-02-11 13:00 +0000",
+			'updated_by': "updated_author",
+			'updated_date': "2015-02-11 07:11 +0000"
+		}
+		assert test_dict != greeting.to_dict()
+
+	def test_to_dict_with_wrong_some_data(self):
+		greeting = Greeting()
+		greeting.author = "test_author"
+		greeting.content = "testing"
+		greeting.date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.updated_by = "updated_author"
+		greeting.updated_date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.put()
+		test_dict = {
+			'id': greeting.key.id(),
+			'author': 'test_author',
+			'content': "testing",
+			'date': "2015-02-11 13:00 +0000",
+			'updated_by': "update_author",  # wrong updated author
+			'updated_date': "2015-02-11 07:11 +0000"
+		}
+		assert test_dict != greeting.to_dict()
+
+	def test_get_greeting_with_no_cursor_in_of_size(self):
+		greetings, next_cursor, next = Greeting.get_greeting(self.guestbook_name, 10, None)
+		assert greetings is not None and next_cursor is not None and next is True
```

Comment:
logic test này không có ý nghĩa, tại sao `cursor==None` thì `next_cursor is not None and next is True` ?